polyfill/lib/calendar.mjs

```diff
@@ -405,161 +1727,203 @@ impl['gregory'] = ObjectAssign({}, impl['iso8601'], {
+    return calendarDate.year;
   },
-
-  isoYear(eraYear, era) {
-    const eraIdx = jpn.eraNames.indexOf(era);
-    if (eraIdx === -1) throw new RangeError(`invalid era ${era}`);
-    return eraYear + jpn.eraAddends[eraIdx];
-  }
-};
-
-impl['japanese'] = ObjectAssign({}, impl['iso8601'], {
-  era(date) {
+  month(date) {
     if (!HasSlot(date, ISO_YEAR) || !HasSlot(date, ISO_MONTH) || !HasSlot(date, ISO_DAY)) {
       date = ES.ToTemporalDate(date, GetIntrinsic('%Temporal.PlainDate%'));
     }
-    return jpn.eraNames[jpn.findEra(date)];
+    const cache = OneObjectCache.getCacheForObject(date);
+    const calendarDate = this.helper.temporalToCalendarDate(date, cache);
+    return calendarDate.month;
   },
-
-  eraYear(date) {
+  day(date) {
     if (!HasSlot(date, ISO_YEAR) || !HasSlot(date, ISO_MONTH) || !HasSlot(date, ISO_DAY)) {
       date = ES.ToTemporalDate(date, GetIntrinsic('%Temporal.PlainDate%'));
     }
-    const eraIdx = jpn.findEra(date);
-    return GetSlot(date, ISO_YEAR) - jpn.eraAddends[eraIdx];
+    const cache = OneObjectCache.getCacheForObject(date);
+    const calendarDate = this.helper.temporalToCalendarDate(date, cache);
+    return calendarDate.day;
   },
-
-  fields(fields) {
-    if (fields.includes('year')) fields.push('eraYear');
-    if (fields.includes('eraYear')) fields.push('era');
-    else if (fields.includes('era')) fields.push('eraYear');
-    return fields;
+  era(date) {
+    if (!HasSlot(date, ISO_YEAR)) date = ES.ToTemporalDate(date, GetIntrinsic('%Temporal.PlainDate%'));
+    if (!this.helper.hasEra) return undefined;
+    const cache = OneObjectCache.getCacheForObject(date);
+    const calendarDate = this.helper.temporalToCalendarDate(date, cache);
+    return calendarDate.era;
   },
+  eraYear(date) {
+    if (!HasSlot(date, ISO_YEAR)) date = ES.ToTemporalDate(date, GetIntrinsic('%Temporal.PlainDate%'));
+    if (!this.helper.hasEra) return undefined;
+    const cache = OneObjectCache.getCacheForObject(date);
+    const calendarDate = this.helper.temporalToCalendarDate(date, cache);
+    return calendarDate.eraYear;
+  },
+  monthCode(date) {
+    if (!HasSlot(date, ISO_MONTH)) date = ES.ToTemporalDate(date, GetIntrinsic('%Temporal.PlainDate%'));
+    const cache = OneObjectCache.getCacheForObject(date);
+    const calendarDate = this.helper.temporalToCalendarDate(date, cache);
+    return calendarDate.monthCode;
+  },
+  dayOfWeek(date) {
+    return impl['iso8601'].dayOfWeek(date);
+  },
+  dayOfYear(date) {
+    const cache = OneObjectCache.getCacheForObject(date);
+    date = ES.ToTemporalDate(date, GetIntrinsic('%Temporal.PlainDate%'));
+    const calendarDate = this.helper.isoToCalendarDate(date, cache);
+    const startOfYear = this.helper.startOfCalendarYear(calendarDate);
+    const diffDays = this.helper.calendarDaysUntil(startOfYear, calendarDate, cache);
+    return diffDays + 1;
+  },
+  weekOfYear(date) {
+    return impl['iso8601'].weekOfYear(date);
```

Comment:
Even without knowing how calendars do weeks, this seems like it will be wrong for some calendars as `ES.WeekOfYear` uses gregorian leap year logic

Reply:
Yeah, week of year is another one of those things that seems really hard to pin down. #1119 and #1058 have discussion about various problems with it. So I essentially punted on it for this PR. I'd prefer to handle in a follow-up. If OK with you, I'll resolve this conversation in favor of those other issues.